### PR TITLE
manifests/apis: more cleanup and automatic generation

### PR DESCRIPTION
**Pull request by joelanford**
This PR is the slice of #234 that involves a few more improvements in the `manifests/apis` directory. The two most substantive changes in this PR are the removal of unnecessary RBAC given to the webhook deployment's service account and the refactoring that enables the validating webhook configuration to be automatically generated from the [kubebuilder markers defined in the bundle webhook implementation](https://github.com/operator-framework/rukpak/blob/3ee8d201ec70315c2dd7080db43771c85884fdb8/api/v1alpha1/bundle_webhook.go#L37)

1. manifests/apis/webhooks: remove all RBAC
2. manifests/apis/webhooks: separate service and certificate out of webhook.yaml
3. manifests/apis/webhooks: generate validatingwebhookconfiguration from kubebuilder markers
4. manifests/apis/crds: remove 'base' directory layer
5. Makefile: remove unnecessary 'controller-gen schemapatch' call

### CODE REVIEW

**Comment by akihikokuroda** [comment]
It’s a lot of changes. I’ll try to check everything Good.

File: manifests/apis/webhooks/resources/service.yaml
Comment (akihikokuroda) [inline manifests/apis/webhooks/resources/service.yaml:7]: This service name and namespace don't match to the client service is defined in the validatingwebhookconfig
```
webhooks:
- admissionReviewVersions:
  - v1
  clientConfig:
    caBundle: ........
    service:
      name: rukpak-webhook-service
      namespace: rukpak-system
      path: /validate-core-rukpak-io-v1alpha1-bundle
      port: 443
```
I got connection refused  message.

**Review by akihikokuroda — APPROVED** [review]
LGTM. Thanks!

File: manifests/apis/webhooks/resources/namespace.yaml
Comment (anik120) [inline manifests/apis/webhooks/resources/namespace.yaml:4]: Why change this name to just `system`? `rukpak-system` was very intuitive to me when I bootstrapped rukpak for the first time. I'd imagine `system` wouldn't be as obvious that that's the namespace where the rukpak resources are deployed.

**Review by anik120 — COMMENTED** [review]
Since we're all learning about kustomization for this repo, would you be open to, very quickly, leave comments explaining some of the features you used for writing the kustomize configs? Example, a short comment on the usage of `vars` and how that's helping us, etc.

Will probably help future devs too to have something like that on record via the PRs

File: manifests/apis/webhooks/resources/namespace.yaml
Comment (akihikokuroda) [reply manifests/apis/webhooks/resources/namespace.yaml:4]: The `namePrefix` in `manifests/apis/webhooks/kustomization.yml` adds `rubpak-` to the names in the webhook/resources.

File: manifests/apis/webhooks/kustomization.yml
Comment (joelanford) [inline manifests/apis/webhooks/kustomization.yml:26]: This defines the variable `CERTIFICATE_NAMESPACE`, which is derived from the `metadata.namespace` value of the `cert-manager.io/v1 Certificate` called `webhook-certificate`.

File: manifests/apis/webhooks/kustomization.yml
Comment (joelanford) [inline manifests/apis/webhooks/kustomization.yml:32]: This defines the variable `CERTIFICATE_NAME`, which is derived from the name of the `cert-manager.io/v1 Certificate` called `webhook-certificate`. 

The object name is the default, used when `fieldref.fieldpath` is not set.

File: manifests/apis/webhooks/kustomization.yml
Comment (joelanford) [inline manifests/apis/webhooks/kustomization.yml:39]: This defines the variable `SERVICE_NAMESPACE`, which is derived from the `metadata.namespace` value of the `v1 Service` called `webhook-service`.

File: manifests/apis/webhooks/kustomization.yml
Comment (joelanford) [inline manifests/apis/webhooks/kustomization.yml:44]: This defines the variable `SERVICE_NAME`, which is derived from the name of the `v1 Service` called `webhook-service`. 

The object name is the default, used when `fieldref.fieldpath` is not set.

File: manifests/apis/webhooks/kustomizeconfig.yaml
Comment (joelanford) [inline manifests/apis/webhooks/kustomizeconfig.yaml:14]: [Name references](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/transformerconfigs/README.md#name-reference-transformer) tie a target resource's name to a list of other resources' referrers' names. Once tied, the referrers' names will change alongside the target name via transformers like namePrefix and nameSuffix.

So the first item in this section says "`spec.issuerRef.name` references the name of a `cert-manager.io Issuer`." This enables kustomize to apply the same transformation to both fields such that a change to an Issuer's name is _also_ applied to a Certificates `spec.issuerRef.name`.

File: manifests/apis/webhooks/kustomizeconfig.yaml
Comment (joelanford) [inline manifests/apis/webhooks/kustomizeconfig.yaml:28]: `varReference` paths are the paths in the specified group/version where variable substitution will occur.

**Comment by joelanford** [comment]
For those reviewing:
1. I rebased this after merging #236 
2.  It'll definitely help to look at this PR commit-by-commit. It's all pretty straightforward except for the commit that switches over to autogenerating the webhook configs from kubebuilder markers.

File: api/v1alpha1/bundle_webhook.go
Comment (timflannagan) [inline api/v1alpha1/bundle_webhook.go:37]: I'm going commit-by-commit when reviewing this PR, so it's possible I'm looking at a backwards facing snapshot: is the `vbundles*` change a typing mistake? Should it be `name=bundles.core.rukpak.io` instead?

File: manifests/apis/webhooks/resources/webhook.yaml
Comment (timflannagan) [inline manifests/apis/webhooks/resources/webhook.yaml:17]: Yep: same comment as above. Is this a spelling mistake?

**Review by timflannagan — APPROVED** [review]
This looks reasonable to me. Thanks for splitting up the commits to make this easier to review. I don't have a strong opinion on moving to a kustomize-focused model, despite kustomize feeling it has a decent barrier of entry at times.

I had one potential blocking comment around the `vbundles` change that I'd like to get clarification on, but I'll approve these changes ahead of time. Do we need a second approval here, or can we push these changes through?

File: api/v1alpha1/bundle_webhook.go
Comment (joelanford) [reply api/v1alpha1/bundle_webhook.go:37]: No that was intentional. I took a hint from the way kubebuilder scaffolds webhooks. It uses `v` and `m` prefixes for validating and mutating webhooks respectively.

**Pull request merged**
